api/worker/handlers.go

```diff
@@ -0,0 +1,78 @@
+package worker
+
+import (
+	"encoding/json"
+	"github.com/ufcg-lsd/arrebol-pb/api"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/auth/token"
+	"net/http"
+)
+
+const SignatureHeader string = "SIGNATURE";
+
+type TokenResponse struct {
+	ArrebolWorkerToken string
+}
+
+func (a *WorkerApi) AddWorker(w http.ResponseWriter, r *http.Request) {
+	var err error
+	signature := r.Header.Get(SignatureHeader)
+
+
+	if string(signature) == "" {
+		api.Write(w, http.StatusBadRequest, api.ErrorResponse{
```

Comment:
I really hate this nested function calls. It is very painful to debug when we need to read the error stacktraces. Is this some sort of idiomatic construct (if so, fine)?